library/math.sol

```diff
@@ -8,4 +9,167 @@ library Math {
+        x = x_a * delta + x_b * delta;
+        y = y_a * delta + y_b * delta;
+        return (x, y);
+    }
+
+    /// @dev Returns the summation of the contents of the array
+    function sum(uint[] toSum) returns (uint s) {
+        uint sum = 0;
+        for (uint i = 0; i < toSum.length; i++){
+            sum += toSum[i];
+        }
+        
+        return sum;
+    }
+    
+    
+    /// @dev Returns difference of list of integers, starting with second argument and subtract all subsequent elements down
+    function diff(uint[] toDiff, uint starting) returns (int){
+        var difference = toDiff[starting];
+        for (uint i = 1; i < toDiff.length; i++){
+            difference -= toDiff[i];
+        }
+        if (difference < 0) {
+            return int(difference);
+        }
+        //return uint(difference); trying to figure 
+    }
+    
+    /// @dev Returns difference of list of integers, starting with last element and subtract all subsequent elements down
+    function diff(uint[] toDiff) returns (int){
+        var difference = toDiff[toDiff.length];
+        for (uint i = 1; i < toDiff.length; i++){
+            difference -= toDiff[i];
+        }
+        if (difference < 0) {
+            return int(difference);
+        }
+        //return uint(difference); trying to figure 
+    }
+    
+    /// @dev calculate factorial of a uint
+    function factor(uint num) returns (uint) {
+        uint o = 1;
+        uint i = 2;
+        while (i <= num){
+            o *= i++;
+        }
+        
+        return o;
+    }
+    
+    /// @dev calculate absolute value of an integer
+    function abs(int num1) returns (int absoluteValue){
+        var n1 = num1;
+        if (n1 < 0) {
+            return n1 * -1;
+        }
+        return n1;
+    }
+    
+    /// @dev returns largest value in array of uints
+    function max(uint[] values) returns (uint maxVal) {
+        uint max = values[0]; 
+        for (uint i = 1; i < values.length; i++){
+            if(values[i] > max){
+                max = values[i];
+            }
+        }
+        return max;
+    }
+
+
+    /// @dev returns smallest value in array of uints
+    function min(uint[] values) returns (uint minVal){
+        uint min = values[0];
+        
+        for (uint i = 0; i < values.length; i++){
+            if (values[i] < min){
+                min = values[i];
+            }
+        }
+        return min;
+    }
+
+    /// @dev returns array filled with range of uints with steps inbetween
+    function range(uint start, uint stop, uint step) returns (uint[] Range) {
+        uint[] memory array;
+        uint i = 0;
+        while (i < stop){
+            array[i++] = start;
+            start += step;
+        }
+    }
+    
+    /// @dev returns array filled with range of uints
+    function range(uint start, uint stop) returns (uint[] Range) {
+        uint[] memory array;
```

Comment:
fair enough...I figured eventually this could have a default value and the third parameter might become unnecessary.